### PR TITLE
feat: Adjust owner_role_type and schema_evolution_record columns

### PR DESCRIPTION
**Pull request by sfc-gh-jmichalak**
- **Support schema evolution record**
- **Support owner role type column**
- **Support owner role type in other views**

## Test Plan

* [ ] acceptance tests

* [x] integration tests

## References

https://docs.snowflake.com/en/user-guide/data-load-schema-evolution

### CODE REVIEW

**Review by sfc-gh-jcieslak — APPROVED** [review]

File: pkg/sdk/testint/helpers_test.go [hunk 1]
```diff
@@ -689,6 +689,7 @@ type informationSchemaColumns struct {
 	IdentityMinimum        sql.NullString `db:"IDENTITY_MINIMUM"`
 	IdentityCycle          sql.NullString `db:"IDENTITY_CYCLE"`
 	IdentityOrdered        sql.NullString `db:"IDENTITY_ORDERED"`
+	SchemaEvolutionRecord  sql.NullString `db:"SCHEMA_EVOLUTION_RECORD"`
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/testint/helpers_test.go:692]: do I see correctly that the name differs than in table? (there are spaces and here underscores) Is it a difference in Snowflake or an error?

File: pkg/sdk/testint/network_rule_gen_integration_test.go [hunk 2]
```diff
@@ -182,4 +182,14 @@ func TestInt_NetworkRulesShowByID(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, id2, e2.ID())
 	})
+
+	t.Run("show by id: check fields", func(t *testing.T) {
+		name := random.AlphaN(4)
+		id1 := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, name)
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/testint/network_rule_gen_integration_test.go:188]: check can be addded to assertValuesAndComment, no need for a separate test

File: pkg/sdk/testint/pipes_integration_test.go [hunk 3]
```diff
@@ -402,4 +402,15 @@ func TestInt_PipesShowByID(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, id2, e2.ID())
 	})
+
+	t.Run("show by id: check fields", func(t *testing.T) {
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/testint/pipes_integration_test.go:406]: can be added to assertPipe, no need for a separate test

File: pkg/sdk/testint/sequences_integration_test.go [hunk 4]
```diff
@@ -215,4 +215,15 @@ func TestInt_SequencesShowByID(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, id2, e2.ID())
 	})
+
+	t.Run("show by id: check fields", func(t *testing.T) {
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/testint/sequences_integration_test.go:219]: can be added to assertSequence, no need for a separate test

File: pkg/sdk/testint/streamlits_integration_test.go [hunk 5]
```diff
@@ -340,4 +340,17 @@ func TestInt_StreamlitsShowByID(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, id2, e2.ID())
 	})
+
+	t.Run("show by id: check fields", func(t *testing.T) {
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/testint/streamlits_integration_test.go:344]: can be added to assertStreamlit, no need for a separate test

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: pkg/sdk/testint/sequences_integration_test.go (hunk 4 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/sequences_integration_test.go]: Done.

File: pkg/sdk/testint/pipes_integration_test.go (hunk 3 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/pipes_integration_test.go]: Done.

File: pkg/sdk/testint/network_rule_gen_integration_test.go (hunk 2 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/network_rule_gen_integration_test.go]: Done.

File: pkg/sdk/testint/tables_integration_test.go [hunk 6]
```diff
@@ -1045,4 +1047,15 @@ func TestInt_TablesShowByID(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, id2, e2.ID())
 	})
+
+	t.Run("show by id: check fields", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/tables_integration_test.go]: Done.

File: pkg/sdk/testint/tables_integration_test.go [hunk 7]
```diff
@@ -500,6 +501,7 @@ func TestInt_Table(t *testing.T) {
 		assertColumns(t, expectedColumns, currentColumns)
 
 		assert.Equal(t, "", table.Comment)
+		assert.Equal(t, sql.NullString{}, currentColumns[2].SchemaEvolutionRecord)
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/tables_integration_test.go]: Done in a separated test case.

File: pkg/sdk/testint/masking_policy_integration_test.go [hunk 8]
```diff
@@ -436,4 +436,15 @@ func TestInt_MaskingPoliciesShowByID(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, id2, e2.ID())
 	})
+
+	t.Run("show by id: check fields", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/masking_policy_integration_test.go:455]: Done.

File: pkg/sdk/testint/helpers_test.go (hunk 1 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/helpers_test.go:208]: This is what I see in `information_schema.columns`. Tests are passing, so I guess it's fine.

File: pkg/sdk/testint/tables_integration_test.go [hunk 9]
```diff
@@ -1014,6 +1014,24 @@ func TestInt_TablesShowByID(t *testing.T) {
 		t.Cleanup(cleanupTableHandle(id))
 	}
 
+	uploadFile := func(t *testing.T, stage *sdk.Stage) string {
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/testint/tables_integration_test.go:1017]: we have a function putOnStage (we had it before in helpers_test.go but now it was moved as part of the helpers refactor and can be accessed by `testClientHelper().Stage.PutOnStage(t, stage.ID(), "<path to file in testdata dir>")`

File: pkg/sdk/testint/tables_integration_test.go [hunk 10]
```diff
@@ -1034,14 +1052,39 @@ func TestInt_TablesShowByID(t *testing.T) {
...
+			sdk.NewAccountObjectIdentifier("ACCOUNTADMIN"),
+			nil)
 		require.NoError(t, err)
-		assert.Equal(t, "ROLE", sl.OwnerRoleType)
+
+		stage, stageCleanup := testClientHelper().Stage.CreateStageInSchema(t, sdk.NewDatabaseObjectIdentifier(testDb(t).Name, schemaTest.Name))
+		t.Cleanup(stageCleanup)
+		filename := uploadFile(t, stage)
+
+		_, err = client.QueryUnsafe(ctx, fmt.Sprintf(`COPY INTO %s
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/testint/tables_integration_test.go:1080]: Please move it to helpers (probably the stage).

Also, isn't ExecForTests more appropriate here? (query expects results, exec just executes)

File: pkg/sdk/testint/tables_integration_test.go [hunk 11]
```diff
@@ -1034,14 +1052,39 @@ func TestInt_TablesShowByID(t *testing.T) {
...
+		FROM @%s/%s
+		FILE_FORMAT = (type=json)
+		MATCH_BY_COLUMN_NAME = CASE_INSENSITIVE`, table.ID().FullyQualifiedName(), stage.ID().FullyQualifiedName(), filename))
+		require.NoError(t, err)
+
+		currentColumns := getTableColumnsFor(t, client, table.ID())
+		require.Len(t, currentColumns, 2)
+		assert.NotEmpty(t, currentColumns[1].SchemaEvolutionRecord)
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/testint/tables_integration_test.go:1088]: is the result changing between the invocations or can we expect a specific value here?

File: pkg/sdk/testint/tables_integration_test.go [hunk 12]
```diff
@@ -1034,14 +1052,39 @@ func TestInt_TablesShowByID(t *testing.T) {
...
+		t.Cleanup(stageCleanup)
+		filename := uploadFile(t, stage)
+
+		_, err = client.QueryUnsafe(ctx, fmt.Sprintf(`COPY INTO %s
+		FROM @%s/%s
+		FILE_FORMAT = (type=json)
+		MATCH_BY_COLUMN_NAME = CASE_INSENSITIVE`, table.ID().FullyQualifiedName(), stage.ID().FullyQualifiedName(), filename))
+		require.NoError(t, err)
+
+		currentColumns := getTableColumnsFor(t, client, table.ID())
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/testint/tables_integration_test.go:1086]: after you merge with main you will have to change this invocation

File: pkg/sdk/testint/tables_integration_test.go [hunk 13]
```diff
@@ -1034,14 +1052,39 @@ func TestInt_TablesShowByID(t *testing.T) {
 		require.Equal(t, id2, e2.ID())
 	})
 
-	t.Run("show by id: check fields", func(t *testing.T) {
+	t.Run("show by id: check schema evolution record", func(t *testing.T) {
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/testint/tables_integration_test.go:1055]: this test checks the new field in information_schema (that differs in name that the one added to `TableColumnDetails` -> result of `DescribeColumns`). Can we check this also? (even in the same test)

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: pkg/sdk/testint/tables_integration_test.go (hunk 9 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/tables_integration_test.go]: Done.

**Review by sfc-gh-asawicki — APPROVED** [review]

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Pull request merged**